src/Domain.LinnApps/Boms/BomChangeService.cs

```diff
@@ -269,37 +272,41 @@ public BomTreeNode CreateBomChanges(BomTreeNode tree, int changeRequestNumber, i
 
         public void CopyBom(string srcPartNumber, string destBomPartNumber, int changedBy, int crfNumber)
         {
-            var destBom = this.bomRepository.FindBy(x => x.BomName == destBomPartNumber);
-            var change = this.bomChangeRepository.FindBy(
-                x => x.DocumentNumber == crfNumber && x.BomId == destBom.BomId && x.ChangeState == "PROPOS");
-            
-            if (change == null)
+            var change = this.GetOrCreateBomChange(destBomPartNumber, crfNumber, changedBy);
+            this.bomPack.CopyBom(srcPartNumber, change.BomId, change.ChangeId, change.ChangeState, "O");
+        }
+
+        public void DeleteAllFromBom(string bomName, int crfNumber, int changedBy)
+        {
+            var change = this.GetOrCreateBomChange(bomName, crfNumber, changedBy);
+            var bom = this.bomRepository.FindBy(b => b.BomName == bomName);
+            foreach (var child in bom.Details)
             {
-                var changeId = this.databaseService.GetIdSequence("CHG_SEQ");
-                change = new BomChange
-                             {
-                                 ChangeId = changeId,
-                                 BomName = destBom.BomName,
-                                 BomId = destBom.BomId,
-                                 DocumentType = "CRF",
-                                 DocumentNumber = crfNumber,
-                                 DateEntered = DateTime.Today,
-                                 EnteredById = changedBy,
-                                 ChangeState = "PROPOS",
-                                 PartNumber = destBom.BomName,
-                                 Comments = "BOM_UT",
-                                 PcasChange = "N"
-                             };
-                this.bomChangeRepository.Add(change);
+                var detail = this.bomDetailRepository.FindById(child.DetailId);
+                if (!detail.DeleteChangeId.HasValue && new[] { "PROPOS", "ACCEPT", "LIVE" }
```

Comment:
what if the changes were added by this change request will this function just delete them or will it mark them as deleted by the same change request i.e. same add and delete change id?

Reply:
yeah good point - as it stands probably the latter which probably isn't right. It should just make them disappear really shouldn't it?